### PR TITLE
[Enhancement](Broker) Add high concurrency broker thrift server.

### PR DESCRIPTION
**Pull request by fmxxxx**
## Proposed changes

This PR has added TThreadedSelectorServer as the default broker thrift server. TThreadedSelectorServer uses non blocking I/O and a multi-threaded model, which is suitable for handling a large number of concurrent connections and improves performance in high concurrency scenarios.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by fmxxxx** [comment]
@WinkerDu @morningman @chenlinzhong Please take a review, thank you.

**Comment by fmxxxx** [comment]
run buildall.

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/common/ThriftServer.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
  */
 public class ThriftServer {
     private static final Logger           LOG  = Logger.getLogger(ThriftServer.class);
-    private              ThriftServerType type = ThriftServerType.THREAD_POOL;
+    private              ThriftServerType type = ThriftServerType.THREADED_SELECTOR;
```
Comment (WinkerDu) [inline fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/common/ThriftServer.java:38]: maybe we can make it configurable for different scenario?

Comment (fmxxxx) [reply fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/common/ThriftServer.java]: sure. done.

**Comment by fmxxxx** [comment]
run buildall.

File: fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/BrokerBootstrap.java [hunk 2]
```diff
@@ -50,7 +50,9 @@ public static void main(String[] args) {
 
             TProcessor tprocessor = new TPaloBrokerService.Processor<TPaloBrokerService.Iface>(
                     new HDFSBrokerServiceImpl());
-            ThriftServer server = new ThriftServer(BrokerConfig.broker_ipc_port, tprocessor);
+            ThriftServer server = new ThriftServer(BrokerConfig.broker_ipc_port,
+                    tprocessor,
+                    ThriftServer.ThriftServerType.valueOf(BrokerConfig.brokerThriftServerType));
```
Comment (WinkerDu) [inline fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/BrokerBootstrap.java:55]: What would happen if the configured server type does not exist?

**Review by WinkerDu — APPROVED** [review]

**Pull request closed without merging**